src/explore-education-statistics-admin/src/pages/release/data/components/DataUploadCancelButton.tsx

```diff
@@ -0,0 +1,44 @@
+import releaseDataFileService from '@admin/services/releaseDataFileService';
+import ButtonText from '@common/components/ButtonText';
+import ModalConfirm from '@common/components/ModalConfirm';
+import useAsyncCallback from '@common/hooks/useAsyncCallback';
+import ErrorMessage from '@common/components/ErrorMessage';
+import useToggle from '@common/hooks/useToggle';
+import React from 'react';
+
+interface Props {
+  releaseId: string;
+  fileId: string;
+}
+
+const DataUploadCancelButton = ({ releaseId, fileId }: Props) => {
+  const [{ error }, cancelImport] = useAsyncCallback(() =>
+    releaseDataFileService.cancelImport(releaseId, fileId),
+  );
+  const [showCancelModal, setShowCancelModal] = useToggle(false);
+  const [showCancelButton, setShowCancelButton] = useToggle(true);
+
+  return (
+    <>
+      {showCancelButton && (
```

Comment:
In the event that the cancellation fails, should we show the button again to allow them to try cancelling again?

Reply:
I'm not sure... if it's failed we don't necessarily want them trying again straight away as it may well just fail again.

Reply:
Okay, we'll keep it removed for now :+1: